sample-app/vault.go

```diff
@@ -95,20 +95,15 @@ func (v *Vault) login(ctx context.Context) (*vault.Secret, error) {
 }
 
 // GetSecretAPIKey fetches the latest version of secret api key from kv-v2
-func (v *Vault) GetSecretAPIKey() (string, error) {
+func (v *Vault) GetSecretAPIKey(ctx context.Context) (string, error) {
 	log.Println("getting secret api key from vault")
 
-	secret, err := v.client.Logical().Read(v.parameters.apiKeyPath)
+	secret, err := v.client.KVv2("kv-v2").Get(ctx, v.parameters.apiKeyPath)
```

Comment:
It might be a good idea to extract "kv-v2" as an environment variable config parameter, maybe `VAULT_API_KEY_MOUNT_PATH`